ul/src/association/client.rs

```diff
@@ -546,11 +570,22 @@ impl<'a> ClientAssociationOptions<'a> {
             user_variables,
         });
 
-        let mut socket = std::net::TcpStream::connect(ae_address)
-            .context(ConnectSnafu)?;
-        socket.set_read_timeout(read_timeout)
+        let mut socket: TcpStream = if let Some(timeout) = connection_timeout {
+            let address = ae_address
+                .to_socket_addrs()
+                .context(ToAddresssSnafu)?
+                .next()
```

Comment:
The default (no connect timeout) would try any of the resolved socket addresses until one is resolved. with this approach, any other socket addresses resolved are ignored in the scenario of one string resolving to multiple socket addresses. To prevent this from being too disruptive, we can just document this behavior in `ClientAssociationOptions::connection_timeout`.

Reply:
Or maybe would you like to mirror the behavior of TcpStream::connection and only fail if all parsed addresses inside the iterator fail to establish connection ?